### ISSUE TITLE
virt-manager can't use zpool for storage(cant refresh volume list)

### ISSUE THREAD

**Issue by ghost** [NixOS/nixpkgs#169387]
### Describe the bug
virt-manager can't use zpool for storage(cant refresh volume list)

### Steps To Reproduce
 `sudo zfs create -V 50G tank/<name>`

### Expected behavior
![2022-04-20_10-23](https://user-images.githubusercontent.com/96685074/164134932-c3b7cb46-5642-4a36-a021-68ebd97c45d0.png)
![2022-04-20_10-24](https://user-images.githubusercontent.com/96685074/164135042-ed0ceb94-dc16-4f44-949b-b78f068eb547.png)

```console
$ zfs list
NAME        USED  AVAIL     REFER  MOUNTPOINT
tank       57.4G   404G      192K  none
tank/home  1.74G   404G     1.74G  legacy
tank/<name>   51.6G   455G       88K  -
tank/nix   4.14G   404G     4.14G  legacy
$ ls /dev/zvol/tank/
lrwxrwxrwx 9 root 20 Apr 10:20 mac -> ../../zd0
```
### Notify maintainers

@globin 

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 5.17.3, NixOS, 22.05 (Quokka), 22.05.20220419.c10c891`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.7.0`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

**Comment by Shawn8901** [comment]
I can so far say that the behavior is for me reproducible. Virt-manager can not see the image list and refresh does also not show infos like "Free storage".
If you need a workaround, if you use "Browse Local" and point that to your zvol, you can create the VM.

**Comment by NIXFOREVERNIX** [comment]
@Shawn8901 Thank you, the workaround works!

**Comment by Shawn8901** [comment]
on my opinion this but was fixed with https://github.com/NixOS/nixpkgs/pull/178485
as the op has deleted the account in the meantime, i can not ask for confirmation, but as i had the same issue and virt-manager can now handle zfs pools correctly for me, i can no longer reproduce the issue, so i am closing it.

**Comment by fadenb** [comment]
@Shawn8901 I just ran into the same issue.
As far as I can tell, the fix from #178485 should be included in the current 22.05 I am using.

I believe the problem stems from another issue (which I, as of now, do not fully understand; Still trying to find out what differs from my older setups):
On the affected system, I do not have a `/dev/zvol` folder (which is referenced in the pool location).

Should we reopen this or better create a new issue?

**Comment by Shawn8901** [comment]
I don't see that this PR was backported. Generally it's that master PRs get a label (so that a Backport is then created by the bot). I also don't find the change on the package https://github.com/MadnessASAP/nixpkgs/blob/d17a56d90ecbd1b8fc908d49598fb854ef188461/pkgs/development/libraries/libvirt/default.nix#L99 . So on my opinion it was not applied to 22.05. You could override the package with the change and use that version as overlay or own package or mix with unstable to confirm if it also fixes your issue. 
If it's fixed then, it's the same issue, if it's not fixed afterwards it's likely something else
